bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/TemperatureChannelHelper.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.handler.channelhelper;
+
+import static org.openhab.binding.netatmo.internal.NetatmoBindingConstants.*;
+import static org.openhab.binding.netatmo.internal.utils.ChannelTypeUtils.*;
+import static org.openhab.binding.netatmo.internal.utils.WeatherUtils.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.MeasureClass;
+import org.openhab.binding.netatmo.internal.api.dto.Dashboard;
+import org.openhab.core.types.State;
+
+/**
+ * The {@link TemperatureChannelHelper} handles specific behavior of modules measuring temperature
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class TemperatureChannelHelper extends ChannelHelper {
+    public TemperatureChannelHelper() {
+        this(GROUP_TEMPERATURE, MeasureClass.INSIDE_TEMPERATURE);
+    }
+
+    protected TemperatureChannelHelper(String groupName, MeasureClass measureClass) {
+        super(groupName, measureClass);
+    }
+
+    @Override
+    protected @Nullable State internalGetDashboard(String channelId, Dashboard dashboard) {
+        switch (channelId) {
+            case CHANNEL_VALUE:
+                return toQuantityType(dashboard.getTemperature(), MeasureClass.OUTSIDE_TEMPERATURE);
+            case CHANNEL_MIN_VALUE:
+                return toQuantityType(dashboard.getMinTemp(), MeasureClass.OUTSIDE_TEMPERATURE);
+            case CHANNEL_MAX_VALUE:
+                return toQuantityType(dashboard.getMaxTemp(), MeasureClass.OUTSIDE_TEMPERATURE);
```

Comment:
For these 3 channels, is it MeasureClass.OUTSIDE_TEMPERATURE or MeasureClass.INSIDE_TEMPERATURE that should be used ?

Reply:
Not really. TemperatureChannelHelper may be used by indoor or outdoor modules. There is no way here to decide.
The usage of MeasureClass.OUTSIDE_TEMPERATURE instead of MeasureClass.INSIDE_TEMPERATURE is by design because OUTSIDE_TEMPERATURE has lower and higher measure limits than INSIDE_TEMPERATURE.

Reply:
Please add a comment with this explanation in the code.
If there are other cases where you did that, please also comment.

Reply:
Done.